--- a/demux/demux_lavf.c	2026-06-08 09:58:55.482546416 -0300
+++ b/demux/demux_lavf.c	2026-06-08 09:58:55.483213483 -0300
@@ -258,7 +258,7 @@
     int num_nested;
     int (*default_io_open)(struct AVFormatContext *s, AVIOContext **pb,
                            const char *url, int flags, AVDictionary **options);
-    void (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
+    int (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
 } lavf_priv_t;
 
 static void update_read_stats(struct demuxer *demuxer)
@@ -930,7 +930,7 @@
     return r;
 }
 
-static void nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
+static int nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
 {
     struct demuxer *demuxer = s->opaque;
     lavf_priv_t *priv = demuxer->priv;
@@ -943,7 +943,7 @@
     }
 
 
-    priv->default_io_close(s, pb);
+    return priv->default_io_close(s, pb);
 }
 
 static int demux_open_lavf(demuxer_t *demuxer, enum demux_check check)
@@ -1041,9 +1041,9 @@
     avfc->opaque = demuxer;
     if (demuxer->access_references) {
         priv->default_io_open = avfc->io_open;
-        priv->default_io_close = avfc->io_close;
+        priv->default_io_close = avfc->io_close2;
         avfc->io_open = nested_io_open;
-        avfc->io_close = nested_io_close;
+        avfc->io_close2 = nested_io_close;
     } else {
         avfc->io_open = block_io_open;
     }
