--- a/demux/demux_lavf.c	2026-06-08 10:10:14.014116604 -0300
+++ b/demux/demux_lavf.c	2026-06-08 10:10:14.015513450 -0300
@@ -258,7 +258,7 @@
     int num_nested;
     int (*default_io_open)(struct AVFormatContext *s, AVIOContext **pb,
                            const char *url, int flags, AVDictionary **options);
-    void (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
+    int (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
 } lavf_priv_t;
 
 static void update_read_stats(struct demuxer *demuxer)
@@ -930,7 +930,7 @@
     return r;
 }
 
-static void nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
+static int nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
 {
     struct demuxer *demuxer = s->opaque;
     lavf_priv_t *priv = demuxer->priv;
@@ -943,7 +943,7 @@
     }
 
 
-    priv->default_io_close(s, pb);
+    return priv->default_io_close(s, pb);
 }
 
 static int demux_open_lavf(demuxer_t *demuxer, enum demux_check check)
@@ -1041,9 +1041,9 @@
     avfc->opaque = demuxer;
     if (demuxer->access_references) {
         priv->default_io_open = avfc->io_open;
-        priv->default_io_close = avfc->io_close;
+        priv->default_io_close = avfc->io_close2;
         avfc->io_open = nested_io_open;
-        avfc->io_close = nested_io_close;
+        avfc->io_close2 = nested_io_close;
     } else {
         avfc->io_open = block_io_open;
     }
--- a/filters/f_lavfi.c	2026-06-08 10:10:14.319111492 -0300
+++ b/filters/f_lavfi.c	2026-06-08 10:10:14.320207151 -0300
@@ -1026,7 +1026,7 @@
     case AV_OPT_TYPE_VIDEO_RATE:        return "fps";
     case AV_OPT_TYPE_DURATION:          return "duration";
     case AV_OPT_TYPE_COLOR:             return "color";
-    case AV_OPT_TYPE_CHANNEL_LAYOUT:    return "channellayout";
+    case AV_OPT_TYPE_CHLAYOUT:    return "channellayout";
     case AV_OPT_TYPE_BOOL:              return "bool";
     case AV_OPT_TYPE_CONST: // fallthrough
     default:
