--- demux/demux_lavf.c
+++ demux/demux_lavf.c
@@ -258,7 +258,7 @@
     int (*default_io_open)(struct AVFormatContext *s, AVIOContext **pb,
                            const char *url, int flags,
                            AVDictionary **options);
-    void (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
+    int (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
 };
 
 static void free_demux(mp_demuxer_t *demuxer)
@@ -930,12 +930,13 @@
     return priv->default_io_open(s, pb, url, flags, options);
 }
 
-static void nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
+static int nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
 {
     lavf_priv_t *priv = s->priv_data;
     if (priv->default_io_close)
-        priv->default_io_close(s, pb);
+        return priv->default_io_close(s, pb);
     av_freep(&pb);
+    return 0;
 }
 
 // Some incredibly stupid protocols, such as rtmp, can be aborted only by
@@ -1041,9 +1042,9 @@
         priv->default_io_open = avfc->io_open;
         avfc->io_open = nested_io_open;
 
-        priv->default_io_close = avfc->io_close;
-        avfc->io_close = nested_io_close;
+        priv->default_io_close = avfc->io_close2;
+        avfc->io_close2 = nested_io_close;
     }
 
     // mpv uses io_open for subtitle/separate audio tracks, which inherits the
