--- a/demux/demux_lavf.c	2026-06-08 10:48:31.818065032 -0300
+++ b/demux/demux_lavf.c	2026-06-08 10:48:31.818652202 -0300
@@ -258,7 +258,7 @@
     int num_nested;
     int (*default_io_open)(struct AVFormatContext *s, AVIOContext **pb,
                            const char *url, int flags, AVDictionary **options);
-    void (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
+    int (*default_io_close)(struct AVFormatContext *s, AVIOContext *pb);
 } lavf_priv_t;
 
 static void update_read_stats(struct demuxer *demuxer)
@@ -930,7 +930,7 @@
     return r;
 }
 
-static void nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
+static int nested_io_close(struct AVFormatContext *s, AVIOContext *pb)
 {
     struct demuxer *demuxer = s->opaque;
     lavf_priv_t *priv = demuxer->priv;
@@ -943,7 +943,7 @@
     }
 
 
-    priv->default_io_close(s, pb);
+    return priv->default_io_close(s, pb);
 }
 
 static int demux_open_lavf(demuxer_t *demuxer, enum demux_check check)
@@ -1041,9 +1041,9 @@
     avfc->opaque = demuxer;
     if (demuxer->access_references) {
         priv->default_io_open = avfc->io_open;
-        priv->default_io_close = avfc->io_close;
+        priv->default_io_close = avfc->io_close2;
         avfc->io_open = nested_io_open;
-        avfc->io_close = nested_io_close;
+        avfc->io_close2 = nested_io_close;
     } else {
         avfc->io_open = block_io_open;
     }
--- a/filters/f_lavfi.c	2026-06-08 10:48:32.104166188 -0300
+++ b/filters/f_lavfi.c	2026-06-08 10:48:32.104406778 -0300
@@ -1026,7 +1026,7 @@
     case AV_OPT_TYPE_VIDEO_RATE:        return "fps";
     case AV_OPT_TYPE_DURATION:          return "duration";
     case AV_OPT_TYPE_COLOR:             return "color";
-    case AV_OPT_TYPE_CHANNEL_LAYOUT:    return "channellayout";
+    case AV_OPT_TYPE_CHLAYOUT:    return "channellayout";
     case AV_OPT_TYPE_BOOL:              return "bool";
     case AV_OPT_TYPE_CONST: // fallthrough
     default:
--- a/video/decode/vd_lavc.c	2026-06-08 10:48:32.376197381 -0300
+++ b/video/decode/vd_lavc.c	2026-06-08 10:48:32.376557751 -0300
@@ -1156,7 +1156,7 @@
     mpi->dts = mp_pts_from_av(ctx->pic->pkt_dts, &ctx->codec_timebase);
 
     mpi->pkt_duration =
-        mp_pts_from_av(ctx->pic->pkt_duration, &ctx->codec_timebase);
+        mp_pts_from_av(ctx->pic->duration, &ctx->codec_timebase);
 
     av_frame_unref(ctx->pic);
 
--- a/video/sws_utils.c	2026-06-08 10:48:32.652941631 -0300
+++ b/video/sws_utils.c	2026-06-08 10:48:32.653113971 -0300
@@ -21,6 +21,7 @@
 #include <libavcodec/avcodec.h>
 #include <libavutil/bswap.h>
 #include <libavutil/opt.h>
+#include <libavutil/pixdesc.h>
 
 #include "config.h"
 
@@ -303,11 +304,11 @@
     int cr_src = mp_chroma_location_to_av(src.chroma_location);
     int cr_dst = mp_chroma_location_to_av(dst.chroma_location);
     int cr_xpos, cr_ypos;
-    if (avcodec_enum_to_chroma_pos(&cr_xpos, &cr_ypos, cr_src) >= 0) {
+    if (av_chroma_location_enum_to_pos(&cr_xpos, &cr_ypos, cr_src) >= 0) {
         av_opt_set_int(ctx->sws, "src_h_chr_pos", cr_xpos, 0);
         av_opt_set_int(ctx->sws, "src_v_chr_pos", cr_ypos, 0);
     }
-    if (avcodec_enum_to_chroma_pos(&cr_xpos, &cr_ypos, cr_dst) >= 0) {
+    if (av_chroma_location_enum_to_pos(&cr_xpos, &cr_ypos, cr_dst) >= 0) {
         av_opt_set_int(ctx->sws, "dst_h_chr_pos", cr_xpos, 0);
         av_opt_set_int(ctx->sws, "dst_v_chr_pos", cr_ypos, 0);
     }
